src/main/java/picard/arrays/VcfToAdpc.java

```diff
@@ -0,0 +1,164 @@
+                final VariantContext context = variants.next();
+                final float gcScore = getFloatAttribute(context, InfiniumVcfFields.GC_SCORE);
+
+                for (final Genotype genotype : context.getGenotypes()) {
+                    final IlluminaGenotype illuminaGenotype = getIlluminaGenotype(genotype, context);
+
+                    final short rawXIntensity = getShortAttribute(genotype, InfiniumVcfFields.X);
+                    final short rawYIntensity = getShortAttribute(genotype, InfiniumVcfFields.Y);
+                    final float normalizedXIntensity = getFloatAttribute(genotype, InfiniumVcfFields.NORMX);
+                    final float normalizedYIntensity = getFloatAttribute(genotype, InfiniumVcfFields.NORMY);
+
+                    final IlluminaAdpcFileWriter.Record record = new IlluminaAdpcFileWriter.Record(rawXIntensity, rawYIntensity, normalizedXIntensity, normalizedYIntensity, gcScore, illuminaGenotype);
+                    adpcRecordList.add(record);
+                }
+            }
+            adpcFileWriter.write(adpcRecordList);
+        } catch (Exception e) {
+            log.error(e);
+            return 1;
+        }
+
+        return 0;
+    }
+
+    private IlluminaGenotype getIlluminaGenotype(final Genotype genotype, final VariantContext context) {
+        IlluminaGenotype illuminaGenotype = picard.arrays.illumina.IlluminaGenotype.NN;
+        if (genotype.isCalled()) {
+            // Note that we remove the trailing '*' that appears in alleles that are reference.
+            final String illuminaAlleleA = StringUtils.stripEnd(getStringAttribute(context, InfiniumVcfFields.ALLELE_A), "*");
+            final String illuminaAlleleB = StringUtils.stripEnd(getStringAttribute(context, InfiniumVcfFields.ALLELE_B), "*");
+            if (genotype.getAlleles().size() != 2) {
+                throw new PicardException("Unexpected number of called alleles in variant context " + context + " found alleles: " + genotype.getAlleles());
+            }
+            final Allele calledAllele1 = genotype.getAllele(0);
+            final Allele calledAllele2 = genotype.getAllele(1);
+
+            if (calledAllele1.basesMatch(illuminaAlleleA)) {
+                if (calledAllele2.basesMatch(illuminaAlleleA)) {
+                    illuminaGenotype = picard.arrays.illumina.IlluminaGenotype.AA;
+                } else if (calledAllele2.basesMatch(illuminaAlleleB)) {
+                    illuminaGenotype = picard.arrays.illumina.IlluminaGenotype.AB;
+                }
+            } else if (calledAllele1.basesMatch(illuminaAlleleB)) {
+                if (calledAllele2.basesMatch(illuminaAlleleA)) {
+                    illuminaGenotype = picard.arrays.illumina.IlluminaGenotype.AB;
+                } else if (calledAllele2.basesMatch(illuminaAlleleB)) {
+                    illuminaGenotype = picard.arrays.illumina.IlluminaGenotype.BB;
+                }
+            }
```

Comment:
convert following if to an else and error out.

Reply:
ok